adapter/internal/oasparser/utills/open_api_utils.go

```diff
@@ -51,25 +52,28 @@ func hasPrefix(buf []byte, prefix []byte) bool {
 	return bytes.HasPrefix(trim, prefix)
 }
 
-// FindSwaggerVersion finds the openapi version ("2" or "3") for the given
+// FindAPIDefinitionVersion finds the openapi version ("2" or "3") for the given
 // openAPI json content.
-func FindSwaggerVersion(jsn []byte) string {
+func FindAPIDefinitionVersion(jsn []byte) string {
 	var version string = "2"
 	var result map[string]interface{}
 
 	err := json.Unmarshal(jsn, &result)
 	if err != nil {
-		logger.LoggerOasparser.Error("json unmarsheliing err when finding the swaggerVersion : ", err)
+		logger.LoggerOasparser.Error("Error while JSON unmarshalling to find the API definition version.", err)
 	}
 
 	if _, ok := result["swagger"]; ok {
-		version = "2"
+		return "2"
 	} else if _, ok := result["openapi"]; ok {
-		version = "3"
-	} else {
-		logger.LoggerOasparser.Warn("swagger file version is not defined. Default version set as to 2 ")
-		return version
-
+		return "3"
+	} else if versionNumber, ok := result["asyncapi"]; ok {
+		if strings.HasPrefix(versionNumber.(string), "2") {
+			return "asyncapi_2"
+		}
+		logger.LoggerOasparser.Errorf("AsyncAPI version %v is not supported.", versionNumber.(string))
+		return "not_supported"
```

Comment:
Shall we return an error? Instead of a string

Reply:
the returned values from this function will be checked and then returned an error
https://github.com/Thushani-Jayasekera/product-microgateway/blob/cb208a27c188f67f5061ae201df6b20add266a28/adapter/internal/oasparser/model/mgw_swagger.go#L1234

Reply:
This is okay, but if possible let's return an error. If there are many changes to be done in the calling functions, let's keep this as it is.